### PR TITLE
hvloader: resolve CVEs in edk2's bundled openssl

### PR DESCRIPTION
**Pull request by arc9693**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This patch addresses following openssl related CVEs affecting hvloader package:
- CVE-2023-0286
- CVE-2023-0215
- CVE-2022-4450
- CVE-2022-4304
 Updates edk2 submodules dependency to edk2-stable202302 tag.

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-0286
- https://nvd.nist.gov/vuln/detail/CVE-2023-0215
- https://nvd.nist.gov/vuln/detail/CVE-2022-4450
- https://nvd.nist.gov/vuln/detail/CVE-2022-4304

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=567158&view=logs&j=bac295d9-f2e0-5165-e63e-e76f383e1a27&t=11f6dde0-e75e-52b2-fbf0-d83a00678fa3

### CODE REVIEW

**Review by 0xba1a — APPROVED** [review]
LGTM

**Review by Camelron — APPROVED** [review]

**Comment by ms-mahuber** [comment]
Can a pointer to end-to-end testing be provided?

File: SPECS-SIGNED/hvloader-signed/hvloader-signed.spec [hunk 1]
```diff
@@ -69,6 +69,9 @@ popd
 /boot/efi/HvLoader.efi
 
 %changelog
+* Fri May 10 2024 Archana Choudhary <archana1@microsoft.com> - 1.0.1-2.cm2
```
Comment (PawelWMS) [inline SPECS-SIGNED/hvloader-signed/hvloader-signed.spec:72]:
```suggestion
* Fri May 10 2024 Archana Choudhary <archana1@microsoft.com> - 1.0.1-2
```

File: SPECS-SIGNED/hvloader-signed/hvloader-signed.spec [hunk 2]
```diff
@@ -6,7 +6,7 @@
 Summary:        Signed HvLoader.efi for %{buildarch} systems
 Name:           hvloader-signed-%{buildarch}
 Version:        1.0.1
-Release:        1%{?dist}
+Release:        2%{?dist}
```
Comment (PawelWMS) [inline SPECS-SIGNED/hvloader-signed/hvloader-signed.spec:9]: Ah, a first "signed" spec I haven't tested to run through fast-track. Fingers crossed.:)

**Pull request merged**